### DISCUSSION TITLE
How do I generate Refresh token using Postman

### DISCUSSION THREAD

**Discussion by TimirC** [etsy/open-api#1095] [Q&A] [answered]
I am using postman to generate the Access token, I followed the documentation and made a new POST request to get the refresh token which will be valid for 90 days. However I am new to this and I am sure I am doing some mistakes here. 

My Authorization is set in the collection section, so all the APIs inherit the Authorization from the Parent Collection.

I am making a POST request at url
https://openapi.etsy.com/v3/public/oauth/token

In the Body I am selecting x-www-form-urlencoded as mentioned in the doc. 

and adding following parameters to it.

grant_type : refresh_code
client_id : MY CLIENT ID
refresh_token : MY REFRESH TOKEN 

On making this call I am getting the following error

{
    "error": "unsupported_grant_type",
    "error_description": "Grant type is not supported"
}

Can anyone please help me understand what am I doing wrong here?
Thank you in Advance

**Comment by seandmccarthy** [comment, marked as answer]
grant_type should be "refresh_token"

**Comment by TimirC** [comment]
Thank you @seandmccarthy ,
I realized that and did the change, I am able to generate new token now, but its validity is still 3600. Is this correct?
or It should be more than this?

**Reply by Gareth064** [reply]
You use the refresh token to get a new access token AND another new refresh token.

The access token will always be 3600. And the refresh token will be whatever it is... A lot longer.

**Reply by TimirC** [reply]
Thank you @Gareth064 for the clarification, 
I referred to your answer [here ](https://github.com/etsy/open-api/discussions/996#discussioncomment-5915404), where it said 86400. My bad, I read from the screenshot only, and not your response. 


What I have learned from both the answers is that, Once I create an Access token, its validity will be 3600, the refresh token which I get along with the access token will have the validity of 1 day. I have to keep recycling the refresh token Every time I make an API call(if the Access token expires) . 

I am making some automation in google sheets using AppScript for my personal use, Is there a way I can simplify this Authentication process? I would really like a method that was simply based on Access Token😅 and which requires authentication only once.

**Reply by Gareth064** [reply]
Unfortunately there is no simpler way to do the OAuth dance. When the v3 API was created, Etsy choose OAuth 2.0 as the only authentication method. Unlike somewhere like Shopify, where if you are building a personal app, you can get a personal access token and use it continually without further auth steps. But only when you want your app to be used by others, then you need to do the OAuth dance.

**Reply by TimirC** [reply]
Got it @Gareth064 ,
Once again thank you so much for your time and efforts.
I will try to build a proper mechanism for the same using Google Sheets and Apps Script, so that as and when the script runs it has the valid Token.

**Reply by PdubsCryptoWorld** [reply]
@Gareth064    I want to make sure i have this right as I find it confusing.

I send my first request and get an access_token and a refresh_token.
The access token is good for 1 hour correct.

However the refresh_token is good for 86400 seconds ie 1 day. Even though it doesn't say.

And then at the end i would just send a refresh_token request and it will give me another set of tokens?

So after i make my first request, I simply just use the refresh token for 24 hours? Afterwards, i send that as the refresh request and get a new set to use? Always using the refresh_token because it's the long lived one?

Thanks in advance.

**Reply by thisisjake** [reply]
@PdubsCryptoWorld from memory it is 90 days, its certainly not 1 day even though the documentation seems to indicate that.
